### PR TITLE
Woop installer: Implement new design / error handling

### LINKED ISSUE #58468: Woo Installer: Implement mobile ux and new design - Confirm Step
### Details

Implement the new design for Confirm Step #58387.
Figma: r6eTqDpL2GGk96hcPLhsBb-fi-431%3A2447

### Info, warnings, and blockers.

image

According to the design above, we need to reorganize the eligibility data in order to apply and handle the different areas in:

* WPCom subdomain warning: first area _New Store Domain_ area
* Generic warnings:  second area _Thins you should be aware of_
* Blockers: third area: error message.

Also, some blockers will be handled before landing here; for instance when the site doesn't have the proper plan.

Eligibility Warnings | Area / Action to take
-----|------
WPCom subdomain | First area
plugins: an array of unsupported plugins that are used on current blog. | Second Area
widgets: an array of unsupported widgets that are used on current blog. | Second Area

Eligibility Errors | Area / Action to take
-------|--------
`TRANSFER_ALREADY_EXISTS` | -> hit the endpoint
`NO_JETPACK_SITES` | Third area
`EXCESSIVE_DISK_SPACE` | Third area
`NO_BUSINESS_PLAN` | -> redirect to landing page
`NO_VIP_SITES` | Third area
`SITE_PRIVATE` | Third area
`SITE_GRAYLISTED` | Third area
`EMAIL_UNVERIFIED` | Third area
`NON_ADMIN_USER` | Third area
`NOT_RESOLVING_TO_WPCOM` | Third area
`NO_SSL_CERTIFICATE` | Third area
`EXCESSIVE_DISK_SPACE` | Third area
`BLOCKED_ATOMIC_TRANSFER` | Third area

### Checklist

- [x] Implement desktop ux
- [x] Implement mobile ux

### Related
New Design PR: #58527
Plan Issue PR: #58581

### PR DESCRIPTION
**Pull request by retrofox**
#### Changes proposed in this Pull Request

Fixes https://github.com/Automattic/wp-calypso/issues/58468

image

#### Testing instructions

*

Related to #

### CODE REVIEW

File: client/components/eligibility-warnings/domain-warning/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const DomainEligibilityWarning = ( {
+	wpcomDomain,
+	stagingDomain,
+}: DomainEligibilityWarningProps ): ReactElement => (
+	<Card title={ __( 'New Store Domain' ) }>
+		<InfoLabel label={ __( 'New' ) }>{ stagingDomain }</InfoLabel>
+		<p>
+			{ `By installing this product your subdomain will change. Your old subdomain (${ wpcomDomain }) will no longer work. You can change it to a custom domain on us at anytime in future.` }
```
Comment (Rolilink) [inline client/components/eligibility-warnings/domain-warning/index.tsx]: Do we need some translation in here ?

File: client/signup/steps/woocommerce-install/confirm/index.tsx [hunk 2]
```diff
@@ -1,133 +1,139 @@
...
-						onClick={ () => {
-							window.location.href = siteUpgrading.checkoutUrl;
-						} }
-					>
-						{ siteUpgrading.checkoutText }
-					</NextButton>
-				</div>
-			</>
+		return (
+			<PlanWarning title={ __( 'Upgrade your plan' ) }>{ siteUpgrading.description }</PlanWarning>
```
Comment (Rolilink) [inline client/signup/steps/woocommerce-install/confirm/index.tsx:82]: Do we need to translate `siteUpgrading.description`? Or is the server responsible for this translation?

**Review by Rolilink — APPROVED** [review]
I just have some questions around i18n for some strings but feel free to merge!

File: client/components/eligibility-warnings/domain-warning/index.tsx (hunk 1 above)
Comment (retrofox) [reply client/components/eligibility-warnings/domain-warning/index.tsx]: 👍 yes, I copied the code from https://github.com/Automattic/wp-calypso/pull/58527. Let me udpate it.

File: client/signup/steps/woocommerce-install/confirm/index.tsx (hunk 2 above)
Comment (retrofox) [reply client/signup/steps/woocommerce-install/confirm/index.tsx:82]: it's [already translated](https://github.com/Automattic/wp-calypso/blob/c79cff3f5b77a6c951581a94ef438c9452c6cd84/client/signup/steps/woocommerce-install/hooks/use-woop-handling/index.ts#L127-L133) in the hook.

File: client/components/eligibility-warnings/domain-warning/index.tsx (hunk 1 above)
Comment (retrofox) [reply client/components/eligibility-warnings/domain-warning/index.tsx]: addressed [here](https://github.com/Automattic/wp-calypso/pull/58719/commits/258d37664da01f5a2bd886b7a1a4afee7a07562d). Thanks, Rolando. it was a nice one.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7011571

Thank you @retrofox for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7011571).

**Pull request merged**